cpp/include/cudf/copying.hpp

```diff
@@ -79,5 +79,64 @@ std::unique_ptr<column> allocate_like(column_view input, size_type size,
  */
 std::unique_ptr<table> empty_like(table_view input_table);
 
+/**
+ * @brief Slices a `column_view` into a set of `column_view`s according to a set of indices.
+ * The returned view's of `input` are constructed from an even number indices where
+ * the `i`th returned `column_view` will view the elements in `input` indicated by the range
+ * `[indices[i], indices[i+1])`.
```

Comment:
I realize I made a mistake in my earlier suggestion.
```suggestion
 * `[indices[2*i], indices[(2*i)+1])`.
```